src/tests/test_magic.py

```diff
@@ -195,6 +195,162 @@ def test_persist_bare(ip):
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table, expected_result",
+    [
+        ("test", [(0, 1, "foo")]),
+        ("author", [(0, "William", "Shakespeare", 1616)]),
+        (
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_no_override(ip, test_table, expected_result):
+    # New saved dataframe length -> 1
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    assert out.result == expected_result
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table, expected_result",
+    [
+        ("test", [(0, 1, "foo")]),
+        ("author", [(0, "William", "Shakespeare", 1616)]),
+        (
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_override(ip, test_table, expected_result):
+    # Previous saved dataframe length -> 2
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=2)
+    ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
+    # New saved dataframe length -> 1
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    # To test it's referring to new saved dataframe
+    assert out.result == expected_result
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table", [("test"), ("author"), ("website"), ("number_table")]
+)
+def test_persist_replace_override_reverted_order(ip, test_table):
+    # Previous saved dataframe length -> 2
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=2)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+
+    # New saved dataframe length -> 1
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
+    out = ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
+
+    assert "Table already exist, maybe use --persist-replace" in str(out.error_in_exec)
+
+
+@pytest.mark.parametrize(
+    "test_table", [("test"), ("author"), ("website"), ("number_table")]
+)
+def test_persist_and_append_use_together(ip, test_table):
+    # Test error message when use --persist and --persist-replace together
```

Comment:
How are we testing `--persist` and `--persist-replace` here? If we are not, let's remove this line. If we are, let's change the test name which implies something different (`--persist` and `--append`)

Reply:
Looks like we didn't cover the test case that `--persist and --persist-replace` used together.

Do you think we should make it pass or just fail it if it's happening?

Reply:
I asked this because that is what is written in the comment which is different than the actual test and the test name.
As for your question, since you [implemented this behaviour](https://github.com/ploomber/jupysql/pull/440/files/f53e7a2765268154bb0f4dc4d6efac688641fb1e#diff-527caf90dd08297f5253e826bd98fea0b6f4fbd44fe9f20472976f77868dac06R353) - we should check this case and expect for this value error `You cannot both replace and append to a dataframe at same time, please only use either of them`.

Reply:
As for your question, I think it's weird behavior but I wouldn't fail it. 
Maybe add a warning and ignore the `--persist`? @idomic @edublancas what do you think?

Reply:
Fix in https://github.com/ploomber/jupysql/pull/440/commits/b99cf0952ce2d429e93815e2ea966ce52218b55c
![Uploading Screenshot 2023-04-29 at 12.20.26 AM.png…]()